4 - AnimalShelter/Code/AnimalShelter.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+//rework enums
+//restore proper list
+
+namespace AnimalShelter.Code
+{
+    public class AnimalShelter
+    {
+        //private List<Animal> _animals = new List<Animal>();
+        private List<Animal> _animals = new List<Animal>();
+        public List<Animal> animals
+        {
+            get { return _animals; }
+            set { _animals = value; }
+        }
```

Comment:
As it stands, you are exposing the entire list and allows the entire list to be set. This seems like data that should not have this type of exposure. Secondly, if you DID want to expose what your list contains, you should not expose the entire list itself as you are. Due to how pointers work, when you return that list anything that changes the list they get now changes your internal list.

In cases like this, you should return a .ToList() of your list so that they can't affect your list as it creates a clone.